src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIInteract.java

```diff
@@ -115,6 +115,32 @@ protected final boolean mineBlock(@NotNull final BlockPos blockToMine, @NotNull
         //get all item drops
         final List<ItemStack> items = BlockPosUtil.getBlockDrops(world, blockToMine, fortune);
 
+        //if ore then add to statistics
+        if (world.getBlockState(blockToMine).getBlock().equals(Blocks.COAL_ORE)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the number of conditional operators (5) used in the expression (maximum allowed 4). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1067)